From b7d96771fa1dc70cc4d9bbaeaf448d70657b02e9 Mon Sep 17 00:00:00 2001
From: Joshua Strobl <joshua@streambits.io>
Date: Mon, 15 Jun 2020 18:05:03 +0300
Subject: [PATCH 1/1] Add use of DH_set0_pqg

We have this in our OpenSSL but it is not detected as it should.
---
 xio-openssl.c | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/xio-openssl.c b/xio-openssl.c
index 132e8ea..b2c055c 100644
--- a/xio-openssl.c
+++ b/xio-openssl.c
@@ -965,17 +965,12 @@ int
          Error("dh2048 setup failed");
          goto cont_out;
       }
-#if HAVE_DH_set0_pqg
       if (!DH_set0_pqg(dh, p, NULL, g)) {
 	      DH_free(dh);
 	      BN_free(p);
 	      BN_free(g);
 	      goto cont_out;
       }
-#else
-      dh->p = p;
-      dh->g = g;
-#endif /* HAVE_DH_set0_pqg */
       if (sycSSL_CTX_set_tmp_dh(*ctx, dh) <= 0) {
          while (err = ERR_get_error()) {
             Warn3("SSL_CTX_set_tmp_dh(%p, %p): %s", *ctx, dh,
-- 
2.27.0

